shell/run_shell_cmd.go

```diff
@@ -46,51 +44,44 @@ func RunTerraformCommandAndCaptureOutput(terragruntOptions *options.TerragruntOp
 func RunShellCommandWithOutput(terragruntOptions *options.TerragruntOptions, command string, args ...string) (string, error) {
 	terragruntOptions.Logger.Printf("Running command: %s %s", command, strings.Join(args, " "))
 
-	var outToErr bool = false
+	var outBuf bytes.Buffer
 
 	cmd := exec.Command(command, args...)
 
-	stdoutIn, err := cmd.StdoutPipe()
-	if err != nil {
-		return "", errors.WithStackTrace(err)
-	}
-
-	stderrIn, err := cmd.StderrPipe()
-	if err != nil {
-		return "", errors.WithStackTrace(err)
-	}
-
 	// TODO: consider adding prefix from terragruntOptions logger to stdout and stderr
 	cmd.Stdin = os.Stdin
 	cmd.Env = toEnvVarsList(terragruntOptions.Env)
 
+	var errWriter = terragruntOptions.ErrWriter
+	var outWriter = terragruntOptions.Writer
 	// Terragrunt can run some commands (such as terraform remote config) before running the actual terraform
 	// command requested by the user. The output of these other commands should not end up on stdout as this
 	// breaks scripts relying on terraform's output.
 	if !reflect.DeepEqual(terragruntOptions.TerraformCliArgs, args) {
-		outToErr = true
+		outWriter = terragruntOptions.ErrWriter
 	}
 
+	stdout := io.MultiWriter(outWriter, &outBuf)
+	stderr := io.MultiWriter(errWriter, &outBuf)
+
 	cmd.Dir = terragruntOptions.WorkingDir
+	cmd.Stderr = stderr
+	cmd.Stdout = stdout
```

Comment:
Could you just set these directly to `io.MultiWriter` without the extra intermediate variables?

Reply:
Also, could you add a link to https://blog.kowalczyk.info/article/wOYk/advanced-command-execution-in-go-with-osexec.html so future maintainers can reference it later?

Reply:
Done.